File: 121-Best-Time-to-Buy-and-Sell-Stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit =0;
        int mini = prices[0];
        for (int stock =0 ; stock < prices.length; stock++){
            mini = Math.min(mini, prices[stock]);
            profit = Math.max(profit,prices[stock] - mini);
        }
        return profit;
    }
}

File: 121-Best-Time-to-Buy-and-Sell-Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0 
        low_buy = prices[0]
        for sell in range(len(prices)):
            low_buy = min(low_buy, prices[sell])
            profit = max(profit, prices[sell] - low_buy)
        return profit